receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -123,19 +124,20 @@ func (m *Metadata) getExtraResources(podRef stats.PodReference, extraMetadataLab
 }
 
 // getContainerID retrieves container id from metadata for given pod UID and container name,
-// returns an error if no container found in the metadata that matches the requirements.
+// returns an error if no container found in the metadata that matches the requirements
+// or if the containerID is empty.
 func (m *Metadata) getContainerID(podUID string, containerName string) (string, error) {
 	uid := types.UID(podUID)
 	for _, pod := range m.PodsMetadata.Items {
 		if pod.UID == uid {
 			for _, containerStatus := range pod.Status.ContainerStatuses {
 				if containerName == containerStatus.Name {
-					return stripContainerID(containerStatus.ContainerID), nil
+					return stripContainerID(containerStatus.ContainerID, podUID, containerName)
 				}
 			}
 			for _, initContainerStatus := range pod.Status.InitContainerStatuses {
 				if containerName == initContainerStatus.Name {
-					return stripContainerID(initContainerStatus.ContainerID), nil
+					return stripContainerID(initContainerStatus.ContainerID, podUID, containerName)
 				}
 			}
```

Comment:
I see you want to reuse `stripContainerID` to avoid copy/pasting the code, but I don't think it's the right function name to do such validation. What if we merge the two loops:

```suggestion
			for _, containerStatus := append(range pod.Status.ContainerStatuses, pod.Status.InitContainerStatuses...) {
				if containerName == containerStatus.Name {
					if len(strings.TrimSpace(id)) == 0 {
						return "", fmt.Errorf("pod %q with container %q has an empty containerID", podUID, containerName)
					}
					return stripContainerID(containerStatus.ContainerID), nil

				}
			}
```

Reply:
yeah, I was thinking of refactoring it. But yeah, I think your suggestion is cleaner. 
I will update the PR

Reply:
code updated with the suggestion above